Better: Add doc for webview SDK. RD-18733 RD-18736 (#43)
ringcentral/engage-digital-api-docs: docs/interactions/structured-messages/webview.md

# Webview

The webview is a convenient way of opening a custom made webpage. It can be chosen by setting the optional target field to `webview` when sending a Structured Message containing a link element, such as [Rich Link](../rich-link), [Carousel](../carousel), or [Template](../template).

The webview is a small window that sits atop the chat.

> # Design

Reviewer: This is EM specific, we don't want this here. I guess it comes from the file move.

Let's do it like JB did on docs/interactions/structured-messages/webview-sdk.md and rename to a Engage Messaging Web example section.

Reviewer 2: It's what I did in 7433c10

<img class="img-fluid" width="419" src="../../../img/web-messaging-webview-portrait.png">

The webview colors can be changed with the `$webview-header-background-color`, and `$webview-header-text-color`. See the [Customization](./customization) page for details.

If you need a smaller window to open, you can also set the `webview_height` parameter to fit your need, you can see the design for `tall` and `compact` webviews in the following visuals.

<img class="img-fluid" width="419" src="../../../img/web-messaging-webview-tall.png">
<img class="img-fluid" width="419" src="../../../img/web-messaging-webview-compact.png">

# Use case

> The webview is not just a simple window opening, it can be used to display a custom survey, to select a room or a seat on a train or even a more attractive way to select a subscription. For each action taken on the page, a custom response can be sent for the agent side, that can confirm a seat selection or express gratitude to a client indicating satisfaction.

Reviewer: Can you say here or maybe below in the Integration section that the custom response / action isn't provided and this is the integrator's job to do it, using ED's API?

Reviewer: With some suggestion like "you can put some identity or converstaion data id in the url query strings"

Reviewer: I guess what I'm asking here is a "good practice" or a "recommended usage" section.

Reviewer 2: Yep good call

Reviewer 2: I added some details, let me know if it is precise enough

Reviewer: it should be enough for now

# Integration

> The page on the webview should be hosted on your side, and setup to accept display on other domains. Check  [Webview SDK](../../structured-messages/webview-sdk) for precisions on the integration.

Reviewer: ```suggestion
The page on the webview should be hosted on your side, and setup to accept display on other domains. Check [Webview SDK](../../structured-messages/webview-sdk) for precisions on the integration.
```

# Webview

> The webview is a convenient way of opening a custom made webpage. It can be chosen by setting the optional target field to `webview` when sending a Structured Message containing a link element, such as [Rich Link](../rich-link), [Carousel](../carousel), or [Template](../template).

Reviewer: ```suggestion
The webview is a convenient way of opening a custom made webpage. It can be chosen by setting the optional `target` field to `webview` when sending a Structured Message containing a link element, such as [Rich Link](../rich-link), [Carousel](../carousel), or [Template](../template).
```

The webview is a small window that sits atop the chat. It can be customized in size through the `webview_height` parameter.

## Use Case

The webview is not just a simple window opening, it can be used to display a custom survey, to select a room or a seat on a train or even a more attractive way to select a subscription. For each action taken on the page, a custom response can be sent for the agent side, that can confirm a seat selection or express gratitude to a client indicating satisfaction.

## Recommended Usage

It should be noted that no default picker or page is provided and this feature is intended to be customizable, not out of the box. No default response after the customer has finished taking actions on the page is provided either.

> What is recommended is setting the conversation's `data-id` inside of the url query string in order to provide a custom page taylored to the customer (e.g displaying a confirmation instead of a choice the second times the customer clicks on it), avoiding him doing several times the procedure.

Reviewer: ```suggestion
What is recommended is setting the conversation's `data-id` inside of the url query string in order to provide a custom page taylored to the customer (e.g., displaying a confirmation instead of a choice the second time the customer clicks on it), avoiding him doing several times the procedure.
```

Reviewer 2: Is the comma after `e.g` intended ?

Reviewer: Yes, `e.g.` means `for example`. So here it's `For example, displaying a confirmation...`

Reviewer 2: Nice I knew it meant `for example` in latin, but I did not know you you needed a comma after using it in an english sentence. I'll add it.

## Channels supporting this feature

* **<span style="color:green">Yes</span>**: Supported
* **<span style="color:orange">Not yet</span>**: Supported by channel but not by Engage Digital yet
* N/A: Not supported by channel

  |  | **webview** | **webview_height** |
  |-|-|-|
  | **Apple Business Chat** | N/A | N/A |
  | **Engage Android Messaging** | **<span style="color:orange">Not yet</span>** | **<span style="color:orange">Not yet</span>** |
  | **Engage iOS Messaging** | **<span style="color:orange">Not yet</span>** | **<span style="color:orange">Not yet</span>** |
  | **Engage Web Messaging** | **<span style="color:green">Yes</span>** | **<span style="color:green">Yes</span>** |
  | **Facebook Messenger** | **<span style="color:orange">Not yet</span>** | **<span style="color:orange">Not yet</span>** |
  | **Google Business Messages** | N/A | N/A |
  | **Source SDK** | **<span style="color:orange">Not yet</span>** | **<span style="color:orange">Not yet</span>** |
  | **Twitter Direct Messages** | **<span style="color:orange">Not yet</span>** | N/A |

>   | **Viber** | **<span style="color:orange">Not yet</span>** | N/A |

Reviewer: We don't really plan on supporting it on Viber, nor on Twitter. So you can N/A them.

  | **WhatsApp** | N/A | N/A |

## Engage Messaging Web specific customization

<img class="img-fluid" width="419" src="../../../img/web-messaging-webview-portrait.png">

The webview colors can be changed with the `$webview-header-background-color`, and `$webview-header-text-color`. See the [Customization](../web-messaging/customization) page for details.

> Engage Messaging is among the source allowing you to set a smaller window to open. To so do, set the `webview_height` parameter to fit your need, you can see the design for `tall` and `compact` webviews in the following visuals.

Reviewer: ```suggestion
Engage Messaging is one of the sources allowing you to set a smaller window to open. To so do, set the `webview_height` parameter to fit your need, you can see the design for `tall` and `compact` webviews in the following visuals.
```
